### PR TITLE
FN1 Checkit

### PR DESCRIPTION
**Pull request by siwelwerd**
Closes https://github.com/TeamBasedInquiryLearning/precalculus/issues/350

### CODE REVIEW

File: precalculus/exercises/outcomes/FN/FN1/generator.sage [hunk 1]
```diff
@@ -1,10 +1,67 @@
...
+        duplicated_x = choice([x for x in pairs_domain if x != pairs_domain[0]])
+        #Make sure we don't put the same ordered pair in there twice.
+        second_y= choice([y for y in pairs_codomain if (duplicated_x,y) not in pairs])                      
+        pairs[0] = (duplicated_x, second_y)
+        shuffle(pairs)
+
+      tasks = [
+                {"map_task": {"map_function": map_function}},
+                {"table_task": {"table_function": table_function,
+                                "table_rows":" ".join([f"<row halign='center'><cell><m>{x}</m></cell><cell><m>{y}</m></cell></row>" for (x,y) in table_pairs])}},
```
Comment (siwelwerd) [inline precalculus/exercises/outcomes/FN/FN1/generator.sage:32]: @StevenClontz I could use some help on getting this table to show up.  It's hard to debug, I don't get any error message it just doesn't show up.

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 2]
```diff
@@ -9,17 +9,89 @@
...
+                    <p>Table goes here.  </p>
+                    <table>
+                    <tabular>
+                        <row header="yes" bottom="minor">
+                            <cell><m>x</m></cell> 
+                            <cell><m>y</m></cell>
+                        </row>
+                        {{table_rows}}
+                    </tabular>
+                </table>
```
Comment (siwelwerd) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:49]: This is the template code for the troublesome table.  I based this off of the code for [Activity 2.1.9(b)](https://teambasedinquirylearning.github.io/precalculus/2024/instructor/FN1.html#FN1-determine-if-a-function-3)

Comment (StevenClontz) [reply precalculus/exercises/outcomes/FN/FN1/template.xml:49]: table/tabular are not supported by SpaTeXt. Instead I think the best solution is to create something in an `<me/>` using LaTeX

Comment (StevenClontz) [reply precalculus/exercises/outcomes/FN/FN1/template.xml:49]:
```xml
            <me>
\begin{array}{c|ccccccc}
x {{#nearbys}}&amp; {{x}}{{/nearbys}} \\\hline
f(x) {{#nearbys}}&amp; \hspace{3em}{{/nearbys}} \\
\end{array}
            </me>
```

https://teambasedinquirylearning.github.io/calculus/2024e/exercises/#/bank/LT2/1/

File: precalculus/exercises/outcomes/FN/FN1/generator.sage (hunk 1 above)
Comment (StevenClontz) [reply precalculus/exercises/outcomes/FN/FN1/generator.sage:32]:
```suggestion
                                "table_rows": [{"x": x, "y": y} for (x,y) in table_pairs],
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml (hunk 2 above)
Comment (StevenClontz) [reply precalculus/exercises/outcomes/FN/FN1/template.xml:49]:
```suggestion
<p>
  <me>
\begin{array}{c|c} 
x &amp; y \\\hline
{{#table_rows}}
{{x}} &amp; {{y}} \\
{{/table_rows}}
\end{array}
  </me>
</p>
```

File: precalculus/exercises/bank.xml [hunk 3]
```diff
@@ -69,6 +69,16 @@
             </description>
         </outcome>  
 
+        <outcome>
+            <title>Introduction to Functions</title>
+            <slug>FN1</slug>
+            <path>outcomes/FN/FN1</path>
+            <description>
+                I can determine if a relation, equation, or graph defines a function
```
Comment (StevenClontz) [inline precalculus/exercises/bank.xml:77]:
```suggestion
                Determine if a relation, equation, or graph defines a function 
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 4]
```diff
@@ -1,26 +1,117 @@
...
-        <content>
-            <p>
-                Task 
-            </p>
-        </content>
-        <outtro>
+        <intro>
             <p>
-                Task answer
+                Explain why each of the following is or is not a function.
```
Comment (StevenClontz) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:7]:
```suggestion
                Explain why each of the following does or does not define a function.
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 5]
```diff
@@ -1,26 +1,117 @@
...
+                        description="Mapping"/>
+                    </p>
+                    </content>
+                    <outtro>
+                        <!-- {{#map_function}}-->
+                        <p>This mapping is a function.</p>
+                        <!-- {{/map_function}}-->
+                        <!-- {{^map_function}}-->
+                        <p>This mapping is not a function.</p>
+                        <!-- {{/map_function}}-->
```
Comment (StevenClontz) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:26]:
```suggestion
                    <outtro>
                        <!-- {{#map_function}}-->
                        <p>This mapping defines a function.</p>
                        <!-- {{/map_function}}-->
                        <!-- {{^map_function}}-->
                        <p>This mapping does not define a function.</p>
                        <!-- {{/map_function}}-->
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 6]
```diff
@@ -1,26 +1,117 @@
...
+                        <!-- {{/pairs_function}}-->
+                    </outtro>
+                </knowl>
+                <!-- {{/pairs_task}} -->
+                <!-- {{/tasks}} -->
+    </knowl>
+
+    <knowl>
+        <intro>
+            <p>Explain why each of the following equations does or does not define a function.</p>
```
Comment (StevenClontz) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:72]:
```suggestion
            <p>Explain why each of the following equations does or does not define a function of <m>x</m>.</p>
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 7]
```diff
@@ -1,26 +1,117 @@
...
+            <knowl>
+                <content>
+                    <p><me>{{eq}}</me></p>
+                </content>
+                <outtro>
+                    <!--{{#eq_function}}-->
+                    <p>This equation is a function.</p>
+                    <!--{{/eq_function}}-->
+                    <!--{{^eq_function}}-->
+                    <p>This equation is not a function.</p>
```
Comment (StevenClontz) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:84]:
```suggestion
                    <p>This equation defines a function of <m>x</m>.</p>
                    <!--{{/eq_function}}-->
                    <!--{{^eq_function}}-->
                    <p>This equation does not define a function of <m>x</m>.</p>
```

File: precalculus/exercises/outcomes/FN/FN1/template.xml [hunk 8]
```diff
@@ -1,26 +1,117 @@
...
+                        source="assets/FN1/generated/{{__seed__}}/{{plot}}.png"
+                        description="Plot"/>
+                </p>
+            </content>
+            <outtro>
+                <!--{{#is_function}}-->
+                <p>This graph is a function.</p>
+                <!--{{/is_function}}-->
+                <!--{{^is_function}}-->
+                <p>This graph is not a function.</p>
```
Comment (StevenClontz) [inline precalculus/exercises/outcomes/FN/FN1/template.xml:109]:
```suggestion
                <p>This graph represents a function.</p>
                <!--{{/is_function}}-->
                <!--{{^is_function}}-->
                <p>This graph does not represent a function.</p>
```

**Review by StevenClontz — APPROVED** [review]

**Comment by StevenClontz** [comment]
approved modulo my suggestions

**Pull request merged**